src/components/appcontent.tsx

```diff
@@ -0,0 +1,81 @@
+import LeaderboardIcon from '../assets/leaderboard.svg';
+import LegendToggleIcon from '../assets/legend_toggle.svg';
+import Navbar from './common/navbar';
+import SearchField from './common/searchfield';
+import SidePaneGrid from './common/sidepanegrid';
+import SidePaneList from './common/sidepanelist';
+
+const MENU_LIST = [
+    { id: 1, name: "Home", icon: DashboardIcon },
+    { id: 2, name: "OPM", icon: LeaderboardIcon },
+    { id: 3, name: "OPM Comparison", icon: DataUsageIcon },
+    { id: 4, name: "Session", icon: LegendToggleIcon },
+    { id: 5, name: "DC Open Orders", icon: DCOpenOrdersIcon }
+]
+
+type Props = {
+    showSidePane: boolean;
+    showNavbar: boolean;
+    appContent: any;
+};
+
+const AppContent: FC<Props> = ({ showSidePane, showNavbar, appContent }) => {
+    const [showSidePaneGrid, setShowSidePaneGrid] = useState<boolean>(false);
+    const [openSearchField, setOpenSearchField] = useState<boolean>(false);
+    const [searchValue, setSearchValue] = useState<string>('');
+    const [selectedMenu, setSelectedMenu] = useState<number>(0);
+
+    return (
+        <div>
+            <div className='h-screen'>
+                {showNavbar &&
+                    <Navbar
+                        showSidePaneGrid={showSidePaneGrid}
+                        setShowSidePaneGrid={setShowSidePaneGrid}
+                        openSearchField={openSearchField}
+                        setOpenSearchField={setOpenSearchField}
+                    />
+                }
+                <div className='flex'>
+                    {showSidePane &&
+                        <SidePaneList
+                            menuList={MENU_LIST}
+                            selectedMenu={selectedMenu}
+                            setSelectedMenu={setSelectedMenu}
+                        />
+                    }
+                    {showSidePane &&
+                        <div className={`${showSidePaneGrid ? `bg-zinc-400` : ''} flex w-full block sm:hidden`} onClick={() => setShowSidePaneGrid(false)}>
+                            {showSidePaneGrid &&
+                                <div>
+                                    <SidePaneGrid
+                                        menuList={MENU_LIST}
+                                        selectedMenu={selectedMenu}
+                                        setSelectedMenu={setSelectedMenu}
+                                    />
+                                </div>}
+                            <div className='block w-full'>
+                                {openSearchField &&
+                                    <SearchField searchValue={searchValue} setSearchValue={setSearchValue}
+                                    />
+                                }
+                                {appContent}
+                            </div>
+                        </div>
+                    }
+
+                    <div className='hidden sm:block'>
```

Comment:
1. Will this be always hidden ?
2. What is usually present in appContent (string or something else ?)

Reply:
1. That's for the web screen and will remain hidden on mobile screens.
2. The entire page excluding header and sidepane